src/crypto/rfc8439.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <crypto/rfc8439.h>
+
+#include <crypto/chacha20.h>
+#include <crypto/common.h>
+
+#include <cstring>
+
+#ifndef RFC8439_TIMINGSAFE_BCMP
+#define RFC8439_TIMINGSAFE_BCMP
+
+int rfc8439_timingsafe_bcmp(const unsigned char* b1, const unsigned char* b2, size_t n)
+{
+    const unsigned char *p1 = b1, *p2 = b2;
+    int ret = 0;
+
+    for (; n > 0; n--)
+        ret |= *p1++ ^ *p2++;
+    return (ret != 0);
+}
+
+#endif // RFC8439_TIMINGSAFE_BCMP
```

Comment:
In commit 7a9d2fb8cf11eef8b6bd87ad1066df374bf3b620: Is there any reason to prefix the define and the function with "rfc8439"? Note that the point here is to let the build system detect if the function `timingsafe_bcmp` is available on the system (if yes, the define `HAVE_TIMINGSAFE_BCMP` is set) and only use the custom implementation if we need to. With this current construct, we would always use the custom one, since `RFC8439_TIMINGSAFE_BCMP` would obviously never be set. This should probably just be like on master (or in bip324_suite.cpp, see other review comment):
```suggestion
#ifndef HAVE_TIMINGSAFE_BCMP
#define HAVE_TIMINGSAFE_BCMP

int timingsafe_bcmp(const unsigned char* b1, const unsigned char* b2, size_t n)
{
    const unsigned char *p1 = b1, *p2 = b2;
    int ret = 0;

    for (; n > 0; n--)
        ret |= *p1++ ^ *p2++;
    return (ret != 0);
}

#endif // HAVE_TIMINGSAFE_BCMP
```

Reply:
Done. Although I did leave it as-is in the second commit and addressed it in the fourth commit because I was getting a linker error for multiple definitions trying it on the second commit. The net change isn't different however.

Reply:
> The net change isn't different however.

No, the net change is you've pointlessly broken things like git bisect. Please try and avoid doing that.

Reply:
@fanquake not sure I understand. At which commit is the build broken? I specifically did it in a way that avoided that but may be misunderstanding something.

Reply:
@dhruv ah, my misread was that you'd left the linker errors between commits 2 & 4. Although, I think introducing this, just to change in later commits is still not ideal. Will re-mark this as resolved, and post new review below.